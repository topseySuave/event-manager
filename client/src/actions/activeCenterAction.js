import axios from 'axios'
import { FETCH_CENTER_DETAIL } from './'
import { showLoading, hideLoading } from 'react-redux-loading-bar'

const fetchCenterDispatch = (data) => {
    return {
        type: FETCH_CENTER_DETAIL,
        center: data
    }
};

export const fetchCenterAction = (id) => {
    if(!id) return 'id is required for the request to be successful';
    return dispatch => {
        dispatch(showLoading());
        return axios.get(`/api/v1/centers/${id}`)
            .then(( { data } )=>{
                dispatch(fetchCenterDispatch(data));
                dispatch(hideLoading());
            })
            .catch((err)=>{
                Materialize.toast('Error in connection!!!', 5000);
                throw (err);
            });
    }
};