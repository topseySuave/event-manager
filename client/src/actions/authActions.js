import axios from 'axios';
import jwtDecode from 'jwt-decode'
import setAuthorizationToken from '../components/authentication/setAuthenticationToken'
import { SET_USER, REMOVE_USER } from './index'

const removeCurrentUser = () => {
    return {
        type: REMOVE_USER,
        payload: {}
    }
};

const setCurrentUser = (token) => {
    return {
        type: SET_USER,
        payload: jwtDecode(token)
    }
};

const signOutRequest = () =>{
    localStorage.removeItem('jwtToken');
    setAuthorizationToken(false);
    return removeCurrentUser();
};

const userSignupRequest = (userData) => {
    // return dispatch => {
        return axios.post('/api/v1/users', userData);
            // .then(res => dispatch(registerUser(res.data)));
    // };
};

const userSignInRequest = (userData) => {
    return dispatch => {
        return axios.post('/api/v1/users/authentication', userData )
            .then(res =>{
                if(res.data.statusCode === 200){
                    const token = res.data.token;
                    localStorage.setItem('jwtToken', token);
                    setAuthorizationToken(token);
                    return dispatch(setCurrentUser(token));
                }else if(res.data.statusCode === 404 || res.data.statusCode === 401){
                    return false;
                }
            });
    };
};

module.exports = {
    userSignupRequest,
    userSignInRequest,
    signOutRequest,
    setCurrentUser
};