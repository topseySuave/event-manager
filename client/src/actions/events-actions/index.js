import { dispatch } from 'redux';
import axios from 'axios';
import {
  FETCH_EVENTS,
  ADD_EVENT,
  EDIT_EVENT,
  EDIT_EVENT_REQUEST,
  EDIT_EVENT_FAILURE,
  REMOVE_EVENT,
  LOADMORE_EVENT_REQUEST,
  LOADMORE_EVENT_SUCCESS,
  LOADMORE_EVENT_FAILURE,
  SEARCH_EVENT_TITLE,
  SESSION_EVENTS,
  SESSION_EVENTS_FAILURE,
  EVENT_STATUS_CHANGE,
  ADD_EVENT_FAILURE
} from '../';
import setAuthorizationToken from
  '../../components/authentication/setAuthenticationToken';


/**
 *  @API Route String
 * * */
const api = '/api/v1/events';

/* *
 *  @Event Dispatch Method
 *  @Returns Object
 * * */
export const eventsDispatchAction = (type, data = {}) => {
  switch (type) {
    case 'edit':
      return {
        type: EDIT_EVENT,
        payload: data
      };

    case 'edit_request':
      return {
        type: EDIT_EVENT_REQUEST,
        payload: data
      };

    case 'add':
      return {
        type: ADD_EVENT,
        payload: data
      };

    case 'fetch':
      return {
        type: FETCH_EVENTS,
        payload: data
      };

    case 'delete':
      return {
        type: REMOVE_EVENT,
        payload: data
      };

    case 'failure':
      return {
        type: ADD_EVENT_FAILURE
      };

    case 'EDIT_EVENT_FAILURE':
      return {
        type: EDIT_EVENT_FAILURE
      };

    case 'SESSION_EVENTS':
      return {
        type: SESSION_EVENTS,
        payload: data
      };

    case 'SESSION_EVENTS_FAILURE':
      return {
        type: SESSION_EVENTS_FAILURE
      };

    case 'EVENT_STATUS_CHANGE':
      return {
        type: EVENT_STATUS_CHANGE
      };

    case 'LOADMORE_EVENT_REQUEST':
      return {
        type: LOADMORE_EVENT_REQUEST
      };

    case 'LOADMORE_EVENT_SUCCESS':
      return {
        type: LOADMORE_EVENT_SUCCESS,
        payload: data
      };

    case 'LOADMORE_EVENT_FAILURE':
      return {
        type: LOADMORE_EVENT_FAILURE
      };

    default:
      return data;
  }
};

export const searchEventsDispatch = data => ({
  type: SEARCH_EVENT_TITLE,
  events: data
});

export const createEvent = (eventData, imgUrl) => (dispatch) => {
  let token = localStorage.getItem('jwtToken') ? localStorage
    .getItem('jwtToken') : false;
  setAuthorizationToken(token);
  eventData.img_url = imgUrl;
  return axios.post(api, eventData)
    .then(({ data }) => {
      if (data.statusCode === 400) {
        Materialize.toast(data.message, 5000, 'red lighten-2');
        dispatch(eventsDispatchAction('failure'));
      } else {
        dispatch(eventsDispatchAction('add', data.event));
        Materialize.toast(data.message, 5000, 'teal lighten-2');
      }
    })
    .catch((err) => {
      dispatch(eventsDispatchAction('EDIT_EVENT_FAILURE'));
      Materialize.toast('An Error Occurred..!!!', 5000, 'red lighten-2');
    });
};

/* *
 *  Initial Create Event Request Action
 *  @Returns Promise
 * * */
export const createEventRequest = eventData => (dispatch) => {
  if (eventData.img_url.name) {
    let formData = new FormData();
    formData.append('file', eventData.img_url);
    formData.append('upload_preset', process.env.CLOUDINARY_UPLOAD_PRESET);
    setAuthorizationToken(false);
    return axios.post(process.env.CLOUDINARY_URL, formData)
      .then(({ data }) => {
        dispatch(createEvent(eventData, data.url));
      })
      .catch((err) => {
        Materialize.toast('Error in connection', 5000, 'red lighten-2');
      });
  }

  return dispatch(createEvent(eventData, ''));
};


/**
 *  @Fetch Event Action
 *  @Returns Object
 * * */
export const fetchEventRequest = () => dispatch => axios.get(api)
  .then(({ data }) => {
    data.loadingmore = false;
    data.loadmore = false;
    data.isLoading = false;
    dispatch(eventsDispatchAction('fetch', data));
  });

export const fetchSessionEventRequest = userId => dispatch => axios
  .get(`${api}?sessionEvents=${userId}`)
  .then(({ data }) => {
    data.isLoading = false;
    if (data) {
      return dispatch(eventsDispatchAction('SESSION_EVENTS', data));
    }
    return dispatch(eventsDispatchAction('SESSION_EVENTS_FAILURE'));
  });

/* *
 *  @Edit Event Request Action
 *  @Returns Object
 * * */
export const editEventRequestAction = data => dispatch =>
  dispatch(eventsDispatchAction('edit_request', data));

const editEvent = (eventData, imgUrl) => (dispatch) => {
  let token = localStorage.getItem('jwtToken') ? localStorage
    .getItem('jwtToken') : false;
  setAuthorizationToken(token);
  eventData.img_url = imgUrl;
  return axios.put(`${api}/${eventData.eventId}`, eventData)
    .then(({ data }) => {
      if (data.statusCode === 201) {
        Materialize.toast(data.message, 5000, 'teal lighten-2');
        $('#add_event_modal').modal('close');
        $('.body__holdr').removeClass('blur__fits');
        return dispatch(eventsDispatchAction('edit', data.event));
      }
      return data;
    });
};

/* *
 *  @Edit Event Action
 *  @Returns Object
 * * */
export const editEventAction = eventData => (dispatch) => {
  if (eventData.img_url.name) {
    let formData = new FormData();
    formData.append('file', eventData.img_url);
    formData.append('upload_preset', process.env.CLOUDINARY_UPLOAD_PRESET);
    setAuthorizationToken(false);
    return axios.post(process.env.CLOUDINARY_URL, formData)
      .then(({ data }) => {
        dispatch(editEvent(eventData, data.url));
      })
      .catch(() => {
        Materialize.toast('Error in connection', 5000, 'red lighten-2');
      });
  }
  return dispatch(editEvent(eventData, eventData.img_url));
};

/* *
 *  @Delete Event Action
 *  @Returns Object
 * * */
export const deleteEventRequest = (id) => {
  id = parseInt(id, 10);
  return dispatch => axios.delete(`${api}/${id}`)
    .then(({ data }) => {
      if (data.statusCode === 200) {
        Materialize.toast(data.message, 5000, 'teal lighten-2');
        return dispatch(eventsDispatchAction('delete', data.event));
      }
      Materialize.toast(data.message, 5000, 'red lighten-2');
      return data;
    });
};

/* *
 *  @Load more Event Request Action
 *  @Returns Object
 * * */
export const loadMoreEvents = offset => (dispatch) => {
  dispatch(eventsDispatchAction('LOADMORE_EVENT_REQUEST'));
  return axios.get(`${api}?next=${offset}`)
    .then(({ data }) => {
      if (data.statusCode === 200) {
        return dispatch(eventsDispatchAction('LOADMORE_EVENT_SUCCESS', data
          .events));
      }
      return dispatch(eventsDispatchAction('LOADMORE_EVENT_FAILURE'));
    });
};

/* *
 * Accept a pending event
 * */
export const handleStatusEventAction = (eventId, status) => dispatch =>
  // send status request for event
  axios.post(`${api}/${eventId}?status=${status}`)
    .then(({ data }) => {
      if (data.statusCode === 200) {
        Materialize.toast(data.message, 5000, 'teal lighten-2');
        return dispatch(eventsDispatchAction('EVENT_STATUS_CHANGE'));
      }
      Materialize.toast(data.message, 5000, 'red lighten-2');
    });

