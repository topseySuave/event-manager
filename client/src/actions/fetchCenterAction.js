import axios from 'axios'
import { FETCH_CENTERS, FETCH_CENTER_DETAIL } from './'
import { showLoading, hideLoading } from 'react-redux-loading-bar'

const fetchCentersDispatch = (data) => {
    return {
        type: FETCH_CENTERS,
        centers: data
    }
};

export const fetchCentersAction = () => {
    return dispatch => {
        dispatch(showLoading());
        return axios.get('/api/v1/centers')
            .then(( { data } )=>{
                dispatch(fetchCentersDispatch(data));
                dispatch(hideLoading());
            })
            .catch((err)=>{
                Materialize.toast('Error in connection!!!', 5000);
                throw (err);
            });
    }
};