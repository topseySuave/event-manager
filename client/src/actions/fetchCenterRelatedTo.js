import axios from 'axios'

export const fetchCenterRelatedTo = ({ location, facilities, capacity, price }) => {
    return axios.get(`/api/v1/centers?search=${location},${price},${facilities},${capacity}&limit=3`);
};