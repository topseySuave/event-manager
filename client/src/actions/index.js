export const ADD_USER = 'ADD_USER';
export const SET_USER = 'SET_USER';
export const REMOVE_USER = 'REMOVE_USER';

export const ADD_CENTER_SUCCESS = 'ADD_CENTER_SUCCESS';
export const ADD_CENTER_FAlLURE = 'ADD_CENTER_FAlLURE';
export const ADD_CENTER_REQUEST = 'ADD_CENTER_REQUEST';
export const FETCH_CENTERS = 'FETCH_CENTERS';
export const FETCH_CENTER_DETAIL = 'FETCH_CENTER_DETAIL';
export const FETCH_RELATED_CENTER = 'FETCH_RELATED_CENTER';

export const FETCH_EVENTS = 'FETCH_EVENTS';
export const ADD_EVENT = 'ADD_CENTER';