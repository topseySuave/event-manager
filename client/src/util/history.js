import createBrowserHistory from 'history/createBrowserHistory';

let history;
export default history = createBrowserHistory();
