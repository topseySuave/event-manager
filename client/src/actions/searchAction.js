import axios from 'axios';
import { isEmpty } from 'lodash';
import queryString from 'query-string';
import {
  fetchCentersDispatch,
  searchCenterDispatch
} from './center-actions/fetchCenterAction';
import { searchEventsDispatch } from './events-actions/index';
import { SEARCH_CENTER_TITLE_FAILED, SEARCH_EVENT_TITLE_FAILED } from './';

const prepareCenterSearchQuery = (searchVal) => {
  let searchObjectString, searchApi, api = '/api/v1/centers?';
  searchObjectString = queryString.stringify(searchVal, {
    arrayFormat: 'bracket'
  });
  searchApi = `${api}${searchObjectString}`;
  return searchApi;
};

const validateEventSearchQuery = ({ searchBy, search }) => {
  let searchApi, api;

  if (searchBy) {
    api = `/api/v1/events?searchBy=${searchBy}&search=`;
  } else {
    api = '/api/v1/events?search=';
  }

  if (!isEmpty(search) && search !== 'undefined') {
    searchApi = `${api + search}`;
  }

  return searchApi;
};

export const searchAction = (data) => {
  let searchApi = prepareCenterSearchQuery(data);
  return dispatch => axios
    .get(searchApi)
    .then(({ data }) => {
      if (data.statusCode === 200) {
        dispatch(fetchCentersDispatch(data));
      } else if (data.statusCode === 404) {
        if (err) {
          Materialize.toast(
            'search result do not match center(s)',
            5000,
            'red'
          );
        }
      }
    })
    .catch((err) => {
      if (err) {
        Materialize.toast(
          'search result do not match center(s)',
          5000,
          'red'
        );
      }
    });
};

export const filterCenterTitle = value => (dispatch) => {
  let searchApi = prepareCenterSearchQuery(value, 'title');
  return axios.get(searchApi).then(({ data }) => {
    if (data.statusCode === 200) {
      dispatch(searchCenterDispatch(data));
    } else if (data.statusCode === 400) {
      Materialize.toast(data.message, 5000, 'red');
      dispatch(searchCenterDispatch(null, 'SEARCH_CENTER_TITLE_FAILED'));
    }
  });
};

export const filterEventTitle = value => (dispatch) => {
  let searchApi = validateEventSearchQuery(value);
  return axios.get(searchApi).then(({ data }) => {
    if (data.statusCode === 200) {
      dispatch(searchEventsDispatch(data.events));
    } else if (data.statusCode === 400) {
      Materialize.toast(data.message, 5000, 'red');
      dispatch({
        type: SEARCH_EVENT_TITLE_FAILED
      });
    }
  });
};
