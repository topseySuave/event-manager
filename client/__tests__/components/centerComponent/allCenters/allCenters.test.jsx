import React from 'react';
import { mount, shallow } from 'enzyme';
import { AllCenters } from
  '../../../../src/components/centerComponent/allCenters/allCenters';
import {
  fetchCentersAction,
  loadMoreCenters
} from '../../../../src/actions/center-actions/fetchCenterAction';
import { searchAction } from '../../../../src/actions/searchAction';
import history from '../../../../src/util/history';

let wrapper, instance, mounted;
let store, props;

const getComponent = () => {
  if (!mounted) {
    props = {
      history: {
        push() {}
      },
      fetchCentersAction,
      searchAction: jest.fn().mockImplementation(() => Promise.resolve()),
      location: {
        search: { location: '' }
      },
      centerStore: {
        centers: [],
        loadingmore: false,
        loadmore: false,
        meta: {
          page: 1,
          pageCount: 2,
          pageSize: 5,
          totalCount: 9
        }
      },
      loadMoreCenters
    };
    mounted = shallow(<AllCenters {...props} />);
  }
  return mounted;
};

describe('AllCenters component', () => {
  beforeEach(() => {
    wrapper = getComponent();
    wrapper.setProps(props);
  });
  let loadMoreSpy = jest.spyOn(AllCenters.prototype, 'loadMore');
  let renderNoCenterSpy = jest.spyOn(AllCenters.prototype, 'renderNoCenter');

  test('that component mounted successfully', () => {
    searchAction({ location: 'ikeja', price: 10000 });
    expect(wrapper).toMatchSnapshot();
  });

  test('component should receive props and update state', () => {
    let newProps = {
      ...props,
      location: {
        search: { location: 'ikeja' }
      }
    };
    wrapper.setProps(newProps);
  });

  test('should be able to search', () => {
    let query = {
      location: 'ikeja',
      price: 200000,
      capacity: 600
    };
    wrapper.instance().onSearch(query);
  });

  test('should be able to load more', () => {
    wrapper.instance().loadMore();
    expect(loadMoreSpy).toBeCalled();
  });

  test('should render no center if there are no centers available', () => {
    wrapper.instance().renderNoCenter();
    expect(renderNoCenterSpy).toBeCalled();
  });
});
