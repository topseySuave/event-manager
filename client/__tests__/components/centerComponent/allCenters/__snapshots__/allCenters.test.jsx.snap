// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllCenters component that component mounted successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllCenters
    centerStore={
      Object {
        "centers": Array [],
        "loadingmore": false,
        "loadmore": false,
        "meta": Object {
          "page": 1,
          "pageCount": 2,
          "pageSize": 5,
          "totalCount": 9,
        },
      }
    }
    fetchCentersAction={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    loadMoreCenters={[Function]}
    location={
      Object {
        "search": Object {
          "location": "",
        },
      }
    }
    searchAction={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="center__holdr"
      >
        <div
          className="row relative"
        >
          <div
            className="col s12 l12"
            style={
              Object {
                "marginBottom": "60px",
              }
            }
          >
            <h4
              className="center-align"
            >
              Boots Centers
            </h4>
            <div
              className="center-align search-faster-form full-width"
            >
              <SearchFasterForm
                onSearch={[Function]}
              />
            </div>
            <div
              className="row"
            >
              <div
                className="col s12 cards-container"
              />
              <h4
                className="bold grey-text lighten-2 center-align"
              >
                <p>
                  No centers Available....
                </p>
              </h4>
              <button
                className="col offset-s3 s6 btn waves-effect gradient__bg"
                onClick={[Function]}
              >
                load more
              </button>
            </div>
          </div>
        </div>
      </div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row relative"
        >
          <div
            className="col s12 l12"
            style={
              Object {
                "marginBottom": "60px",
              }
            }
          >
            <h4
              className="center-align"
            >
              Boots Centers
            </h4>
            <div
              className="center-align search-faster-form full-width"
            >
              <SearchFasterForm
                onSearch={[Function]}
              />
            </div>
            <div
              className="row"
            >
              <div
                className="col s12 cards-container"
              />
              <h4
                className="bold grey-text lighten-2 center-align"
              >
                <p>
                  No centers Available....
                </p>
              </h4>
              <button
                className="col offset-s3 s6 btn waves-effect gradient__bg"
                onClick={[Function]}
              >
                load more
              </button>
            </div>
          </div>
        </div>,
        "className": "center__holdr",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col s12 l12"
            style={
              Object {
                "marginBottom": "60px",
              }
            }
          >
            <h4
              className="center-align"
            >
              Boots Centers
            </h4>
            <div
              className="center-align search-faster-form full-width"
            >
              <SearchFasterForm
                onSearch={[Function]}
              />
            </div>
            <div
              className="row"
            >
              <div
                className="col s12 cards-container"
              />
              <h4
                className="bold grey-text lighten-2 center-align"
              >
                <p>
                  No centers Available....
                </p>
              </h4>
              <button
                className="col offset-s3 s6 btn waves-effect gradient__bg"
                onClick={[Function]}
              >
                load more
              </button>
            </div>
          </div>,
          "className": "row relative",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="center-align"
              >
                Boots Centers
              </h4>,
              <div
                className="center-align search-faster-form full-width"
              >
                <SearchFasterForm
                  onSearch={[Function]}
                />
              </div>,
              <div
                className="row"
              >
                <div
                  className="col s12 cards-container"
                />
                <h4
                  className="bold grey-text lighten-2 center-align"
                >
                  <p>
                    No centers Available....
                  </p>
                </h4>
                <button
                  className="col offset-s3 s6 btn waves-effect gradient__bg"
                  onClick={[Function]}
                >
                  load more
                </button>
              </div>,
            ],
            "className": "col s12 l12",
            "style": Object {
              "marginBottom": "60px",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Boots Centers",
                "className": "center-align",
              },
              "ref": null,
              "rendered": "Boots Centers",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <SearchFasterForm
                  onSearch={[Function]}
                />,
                "className": "center-align search-faster-form full-width",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onSearch": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="col s12 cards-container"
                  />,
                  <h4
                    className="bold grey-text lighten-2 center-align"
                  >
                    <p>
                      No centers Available....
                    </p>
                  </h4>,
                  <button
                    className="col offset-s3 s6 btn waves-effect gradient__bg"
                    onClick={[Function]}
                  >
                    load more
                  </button>,
                ],
                "className": "row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                    "className": "col s12 cards-container",
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      No centers Available....
                    </p>,
                    "className": "bold grey-text lighten-2 center-align",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "No centers Available....",
                    },
                    "ref": null,
                    "rendered": "No centers Available....",
                    "type": "p",
                  },
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "load more",
                    "className": "col offset-s3 s6 btn waves-effect gradient__bg",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "load more",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="center__holdr"
        >
          <div
            className="row relative"
          >
            <div
              className="col s12 l12"
              style={
                Object {
                  "marginBottom": "60px",
                }
              }
            >
              <h4
                className="center-align"
              >
                Boots Centers
              </h4>
              <div
                className="center-align search-faster-form full-width"
              >
                <SearchFasterForm
                  onSearch={[Function]}
                />
              </div>
              <div
                className="row"
              >
                <div
                  className="col s12 cards-container"
                />
                <h4
                  className="bold grey-text lighten-2 center-align"
                >
                  <p>
                    No centers Available....
                  </p>
                </h4>
                <button
                  className="col offset-s3 s6 btn waves-effect gradient__bg"
                  onClick={[Function]}
                >
                  load more
                </button>
              </div>
            </div>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row relative"
          >
            <div
              className="col s12 l12"
              style={
                Object {
                  "marginBottom": "60px",
                }
              }
            >
              <h4
                className="center-align"
              >
                Boots Centers
              </h4>
              <div
                className="center-align search-faster-form full-width"
              >
                <SearchFasterForm
                  onSearch={[Function]}
                />
              </div>
              <div
                className="row"
              >
                <div
                  className="col s12 cards-container"
                />
                <h4
                  className="bold grey-text lighten-2 center-align"
                >
                  <p>
                    No centers Available....
                  </p>
                </h4>
                <button
                  className="col offset-s3 s6 btn waves-effect gradient__bg"
                  onClick={[Function]}
                >
                  load more
                </button>
              </div>
            </div>
          </div>,
          "className": "center__holdr",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col s12 l12"
              style={
                Object {
                  "marginBottom": "60px",
                }
              }
            >
              <h4
                className="center-align"
              >
                Boots Centers
              </h4>
              <div
                className="center-align search-faster-form full-width"
              >
                <SearchFasterForm
                  onSearch={[Function]}
                />
              </div>
              <div
                className="row"
              >
                <div
                  className="col s12 cards-container"
                />
                <h4
                  className="bold grey-text lighten-2 center-align"
                >
                  <p>
                    No centers Available....
                  </p>
                </h4>
                <button
                  className="col offset-s3 s6 btn waves-effect gradient__bg"
                  onClick={[Function]}
                >
                  load more
                </button>
              </div>
            </div>,
            "className": "row relative",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4
                  className="center-align"
                >
                  Boots Centers
                </h4>,
                <div
                  className="center-align search-faster-form full-width"
                >
                  <SearchFasterForm
                    onSearch={[Function]}
                  />
                </div>,
                <div
                  className="row"
                >
                  <div
                    className="col s12 cards-container"
                  />
                  <h4
                    className="bold grey-text lighten-2 center-align"
                  >
                    <p>
                      No centers Available....
                    </p>
                  </h4>
                  <button
                    className="col offset-s3 s6 btn waves-effect gradient__bg"
                    onClick={[Function]}
                  >
                    load more
                  </button>
                </div>,
              ],
              "className": "col s12 l12",
              "style": Object {
                "marginBottom": "60px",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Boots Centers",
                  "className": "center-align",
                },
                "ref": null,
                "rendered": "Boots Centers",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <SearchFasterForm
                    onSearch={[Function]}
                  />,
                  "className": "center-align search-faster-form full-width",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "onSearch": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="col s12 cards-container"
                    />,
                    <h4
                      className="bold grey-text lighten-2 center-align"
                    >
                      <p>
                        No centers Available....
                      </p>
                    </h4>,
                    <button
                      className="col offset-s3 s6 btn waves-effect gradient__bg"
                      onClick={[Function]}
                    >
                      load more
                    </button>,
                  ],
                  "className": "row",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [],
                      "className": "col s12 cards-container",
                    },
                    "ref": null,
                    "rendered": Array [],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <p>
                        No centers Available....
                      </p>,
                      "className": "bold grey-text lighten-2 center-align",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "No centers Available....",
                      },
                      "ref": null,
                      "rendered": "No centers Available....",
                      "type": "p",
                    },
                    "type": "h4",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "load more",
                      "className": "col offset-s3 s6 btn waves-effect gradient__bg",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "load more",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
